kernel/interrupt.rs

```diff
@@ -5,19 +5,33 @@ use crate::{
     net::process_packets,
 };
 use alloc::boxed::Box;
-use alloc::collections::BTreeMap;
 
-// TODO: Use a simple array for faster access.
-static IRQ_HANDLERS: SpinLock<BTreeMap<u8, Box<dyn FnMut() + Send + Sync>>> =
-    SpinLock::new(BTreeMap::new());
+static IRQ_HANDLERS: SpinLock<[Option<Box<dyn FnMut() + Send + Sync>>; 255]> = SpinLock::new([
```

Comment:
An IRQ number can be in the range of [0, 255]. We need 256 slots.

```suggestion
static IRQ_HANDLERS: SpinLock<[Option<Box<dyn FnMut() + Send + Sync>>; 256]> = SpinLock::new([
```

Reply:
How about using a const to avoid writing a lot of `None`?

```
const DEFAULT_IRQ_HANDLER: Option<Box<dyn FnMut() + Send + Sync>> = None;
static IRQ_HANDLERS: SpinLock<[Option<Box<dyn FnMut() + Send + Sync>>; 256]> =
    SpinLock::new([DEFAULT_IRQ_HANDLER; 256]);
```

Reply:
I was googling for solution like this I somehow did not found it :)

Replaced

Reply:
I found [this stackoverflow answer](https://stackoverflow.com/a/66776497). I was beginning to give up on that too ;)